internal/handler/userHandler.go

```diff
@@ -82,24 +94,34 @@ func (h *Handler) Save(context *gin.Context) {
 // @Description Find user by provided id
 // @Tags Users
 // @Produce json
-// @Param id path int true "User id"
-// @Success 201 {object} response.UserResponseDto "User Found successfully"
-// @Failure 422 {object} response.ErrorDto "Id in the wrong format"
+// @Param id path string true "User id"
+// @Failure 422 {object} response.ErrorDto "ID in the wrong format"
 // @Failure 404 {object} response.ErrorDto "User not found"
+// @Failure 500 {object} response.ErrorDto "Failed to connect to database"
 // @Router /users/{id} [get]
 func (h *Handler) FindById(context *gin.Context) {
 	id := context.Param("id")
 	logger.Infof("Finding user by id %s", id)
 	user, err := h.service.GetUserById(context, id)
-	if err != nil {
+	if user == nil {
+		var databaseConnectionErr *exceptions.DatabaseConnectionErr
+		if errors.As(err, &databaseConnectionErr) {
+			logger.Errorf("Error while connecting to database. %v", err)
+			errorResponse := response.ErrorDto{
+				Message:   "Something went wrong",
+				Timestamp: time.Now(),
+			}
```

Comment:
The DatabaseConnectionError is not necessary, it's just a level of indirection. Now that the repository returns nil if a user is not found it does not matter what kind of error we get since we can't handle it. We should return the Mongo error with no midification from the repository. We would then log it and return a 500 status with a generic message in this controller.

Reply:
My comment about the repository should cover this.